src/test/ui/consts/miri_unleashed/mutable_const.rs

```diff
@@ -0,0 +1,20 @@
+// compile-flags: -Zunleash-the-miri-inside-of-you
+
+#![feature(const_raw_ptr_deref)]
+#![deny(const_err)]
+
+use std::cell::UnsafeCell;
+
+// make sure we do not just intern this as mutable
+const MUTABLE_BEHIND_RAW: *mut i32 = &UnsafeCell::new(42) as *const _ as *mut _;
+
```

Comment:
Huh, this compiles? I would have assumed const qualif to complain

Reply:
This is an "unleash Miri" test. ;)

Reply:
This is under `miri_unleashed`, I'm assuming const-checking wouldn't error.

But also... if `miri_unleashed` *isn't* broken, then this should error as leaking the address of a local, no?

Unless the "outermost scope is `'static`" logic propagates through casts.

Reply:
> But also... if miri_unleashed isn't broken, then this should error as leaking the address of a local, no?

Not sure where that error should come from. The local is not destroyed before computation ends (there is no `StorageDead`).

Reply:
But it does error if you use `id(&UnsafeCell::new(42))`, right? Then my last theory is correct (casts are transparent wrt to the scope thing).

Reply:
> This is an "unleash Miri" test. ;)

I knew that, I just assumed there'd be a warning about skipping static checks

Reply:
> I knew that, I just assumed there'd be a warning about skipping static checks

There is: `WARN skipping const checks`

Reply:
not on the `MUTABLE_BEHIND_RAW` constant

Reply:
> But it does error if you use id(&UnsafeCell::new(42)), right? Then my last theory is correct (casts are transparent wrt to the scope thing).

Then we get
```
error: any use of this value will cause an error
  --> src/test/ui/consts/miri_unleashed/mutable_const.rs:11:1
   |
11 | const MUTABLE_BEHIND_RAW: *mut i32 = id(&UnsafeCell::new(42)) as *const _ as *mut _;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered dangling pointer in final constant
   |
```

Reply:
> not on the MUTABLE_BEHIND_RAW constant

Yeah seems like we only get that on uses of constants, but that is consistent with your tests.